xray/commands/audit/scan.go

```diff
@@ -206,19 +209,22 @@ func (scanCmd *ScanCommand) createIndexerHandlerFunc(file *spec.File, indexedFil
 				return err
 			}
 			// In case of empty graph returned by the indexer,
-			// for instance due to unsupported file format, continue without sendding a
+			// for instance due to unsupported file format, continue without sending a
 			// graph request to Xray.
 			if graph.Id == "" {
 				return nil
 			}
-			// Add a new task to the seconde prodicer/consumer
+			// Add a new task to the second producer/consumer
 			// which will send the indexed binary to Xray and then will store the received result.
 			taskFunc := func(threadId int) (err error) {
 				scanResults, err := scanCmd.getXrScanGraphResults(graph, file)
 				if err != nil {
 					return err
 				}
-				resultsArr[threadId] = append(resultsArr[threadId], scanResults)
+				scanErr := commands.CheckScanResultsStatus(scanResults.ScannedStatus , filePath)
+				if scanErr == nil {
+					resultsArr[threadId] = append(resultsArr[threadId], scanResults)
+				}
```

Comment:
I this that if scanErr is not nil, we should add an error log message with the error string. Otherwise, no one will ever know that scanning this component failed.